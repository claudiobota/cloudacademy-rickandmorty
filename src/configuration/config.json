{
	"api": {
		"url": "https://rickandmortyapi.com/api",
		"paths": {
			"characters": "/character",
			"locations": "/location",
			"episodes": "/episode"
		},
		"page_size": 20
	}
}
