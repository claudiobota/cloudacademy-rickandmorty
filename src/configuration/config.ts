import configData from './config.json';
import configDevelopmentData from './config.development.json';
import configTestData from './config.test.json';
import configStagingData from './config.staging.json';
import configProductionData from './config.production.json';
import {IConfig} from '../types/interfaces';
import {Environment} from '../types/enums';

let config: IConfig = {
  ...configData
};

const nodeEnv: string = process.env.NODE_ENV || 'development';

switch (nodeEnv) {
  case Environment.DEVELOPMENT:
    config = {
      ...config,
      ...configDevelopmentData
    };
    break;

  case Environment.TEST:
    config = {
      ...config,
      ...configTestData
    };
    break;

  case Environment.STAGING:
    config = {
      ...config,
      ...configStagingData
    };
    break;

  case Environment.PRODUCTION:
    config = {
      ...config,
      ...configProductionData
    };
    break;

  default:
}

export default config;
