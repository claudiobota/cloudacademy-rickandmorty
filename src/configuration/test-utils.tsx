import React from 'react';
import {render as rtlRender} from '@testing-library/react';
import {Provider} from 'react-redux';
import {configureAppStore} from '../redux/store';

function render(
  ui: React.ReactElement,
  {
    initialState,
    store = configureAppStore(initialState),
    ...renderOptions
  }: any = {}
) {
  function Wrapper({ children }: { children: React.ReactElement }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render }
