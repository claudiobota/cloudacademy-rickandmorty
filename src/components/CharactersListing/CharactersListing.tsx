import React, {useEffect} from 'react';
import {useDispatch, useSelector} from 'react-redux';
import {StoreState} from '../../types/redux-states';
import CharacterItem from '../CharacterItem/CharacterItem';
import {characterActions, fetchCharacters, selectAllCharacters} from '../../redux/slices/characterSlice';
import {Paginator} from 'primereact/paginator';
import config from '../../configuration/config';
import {ICharacter} from '../../types/interfaces';

interface CharactersListingProps {
  characters: ICharacter[];
  hasPagination?: boolean;
  onPageChange?: (page: number) => void;
  page?: number;
  count?: number;
}
export default function CharactersListing(props: CharactersListingProps) {
  const count = props.count || 0;
  const page = props.page || 1;
  const onPageChange = props.onPageChange || ((page: number) => {});
  const characters = props.characters || [];
  const hasPagination = props.hasPagination === true;

  return (
    <div className="wrapper">
      {hasPagination && <Paginator className="p-ml-auto p-mr-auto p-mb-5" rows={config.api.page_size} first={config.api.page_size * (page - 1)} totalRecords={count} onPageChange={e => onPageChange(e.page + 1)} />}
      <div className="p-grid p-align-stretch">
        {characters.map((character, index) => {
          return (
            <div key={index} className="p-col-12 p-sm-4 p-mb-3 p-d-flex p-flex-column">
              <CharacterItem character={character} />
            </div>
          );
        })}
      </div>
      {hasPagination && <Paginator className="p-ml-auto p-mr-auto p-mt-5" rows={config.api.page_size} first={config.api.page_size * (page - 1)} totalRecords={count} onPageChange={e => onPageChange(e.page + 1)} />}
    </div>
  )
}
