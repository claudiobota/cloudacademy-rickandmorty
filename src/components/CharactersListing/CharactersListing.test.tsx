import {render} from '../../configuration/test-utils';
import React from 'react';
import CharactersListing from './CharactersListing';

describe('CharactersListing', () => {
  test('component renders and matches snapshot', () => {
    const { asFragment } = render(<CharactersListing /> );
    expect(asFragment()).toMatchSnapshot();
  });
});
