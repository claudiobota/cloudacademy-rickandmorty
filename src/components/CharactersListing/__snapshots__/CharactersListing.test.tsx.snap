// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersListing component renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="p-grid p-align-stretch"
    />
  </div>
</DocumentFragment>
`;
